### ISSUE TITLE
create-component depends on map key order?

### ISSUE THREAD

**Issue by stuarthalloway** [cljfx/cljfx#117]
`create-component` appears to be sensitive to map key order, e.g. if the `:show` key appears in a map literal before the `:modality`, modality is set after the dialog is shown, causing an `IllegalStateException` on the UI thread.

[Example](https://github.com/stuarthalloway/cljfx/commit/a422c606ad7126a8b4b42cdb06b95bd93f4b3b72)

**Comment by vlaaad** [comment]
Hi @stuarthalloway!

Thank you for the bug report and a repro. I released `1.7.13` that fixes the issue.

The problem is not in `create-component` — it iterates on maps in whatever order the keys are. The problem is that in some cases JavaFX has an order dependency between parts of a single object. For example, window modality has to be set before the window is shown. Unfortunately, cljfx lacks a good way to express the order of mutations. There is a way to do it: `(fx/make-ext-with-props cljfx.fx.text-input-dialogs/props)` will create an extension lifecycle with props that are applied after all the props of its wrapped components are applied. This is clunky for sure and given that it's rare in JavaFX to have such order dependency, I implemented a hack that sets the order of props on a per-component type basis. It was set on other window components such as `:stage` and `:dialog`, but I forgot about those dialog subclasses.
